src/library/Connect/WorkloadInformationProvider.cs

```diff
@@ -117,6 +122,31 @@ internal class WorkloadInformationProvider : IWorkloadInformationProvider
             }
         }
 
+        public List<int> GetPortsToIgnoreFromAnnotations(V1ServiceList serviceNameSpace) {
+            var portsToIgnore = new List<int>();
+
+            List<V1Service> portsList = serviceNameSpace.Items.Where(item => item.Metadata != null && 
+            item.Metadata.Annotations != null &&
+            item.Metadata.Annotations.ContainsKey(ServiceAnnotations)).ToList();

```

Comment:
nit: I believe current code style uses ? for null checks. So something like where(item => item.Metadata?.Annotations?.ContainsKey). More details on this operator [here](https://learn.microsoft.com/en-us/dotnet/csharp/language-reference/operators/member-access-operators#null-conditional-operators--and-)